### PR TITLE
Add OpenBSM support for FreeBSD.

### PR DESCRIPTION
**Pull request by lxcode**
This is copied wholesale from Darwin, save for adding includes for socket.h and in.h. Someday we should figure out a way to deduplicate.

### CODE REVIEW

**Comment by muffins** [comment]
ok to test

**Comment by osqueryer** [comment]
:no_good: The commit 033f5ee6af7612d04b9f75c7aeb1a07087c70250 (Job results: [2481](https://jenkins.osquery.io/job/osqueryPullRequestAudit/2481/)) failed the code audit or documentation test.

**Comment by theopolis** [comment]
Hey @lxcode, do you mind landing the suggested changes:
```diff
diff --git a/osquery/events/freebsd/openbsm.cpp b/osquery/events/freebsd/openbsm.cpp
index 416ab8b..7c5f147 100644
--- a/osquery/events/freebsd/openbsm.cpp
+++ b/osquery/events/freebsd/openbsm.cpp
@@ -15,7 +15,6 @@
 
 #include "osquery/events/darwin/openbsm.h"
 
-
 namespace osquery {
 
 /// The OpenBSM subsystem may have a performance impact on the system.
diff --git a/osquery/tables/events/freebsd/openbsm_events.cpp b/osquery/tables/events/freebsd/openbsm_events.cpp
index 926e7ff..91634e3 100644
--- a/osquery/tables/events/freebsd/openbsm_events.cpp
+++ b/osquery/tables/events/freebsd/openbsm_events.cpp
@@ -8,8 +8,8 @@
  *
  */
 #include <arpa/inet.h>
-#include <sys/socket.h>
 #include <netinet/in.h>
+#include <sys/socket.h>
 
 #include <bsm/libbsm.h>
 ```

**Comment by osqueryer** [comment]
:-1: The commit f8ce78b56962426589ede2a5847f3a0d9caaacbc (Job results: [2722](https://jenkins.osquery.io/job/osqueryPullRequestWindows/2722/)) failed one or more tests (**Windows**).

File: osquery/tables/events/freebsd/openbsm_events.cpp [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+  Status Callback(const OpenBSMEventContextRef& ec,
+                  const OpenBSMSubscriptionContextRef& sc);
+};
+
+REGISTER(OpenBSMExecVESubscriber, "event_subscriber", "process_events");
+REGISTER(OpenBSMSSHLoginSubscriber, "event_subscriber", "user_events");
+
+void OpenBSMExecVESubscriber::configure() {
+  std::vector<size_t> event_ids{
```
Comment (obelisk) [inline osquery/tables/events/freebsd/openbsm_events.cpp]: Are there other ids that should be added here that are specific to FreeBSD?

File: osquery/tables/events/freebsd/openbsm_events.cpp [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+  return Status(0, "OK");
+}
+
+void OpenBSMSSHLoginSubscriber::configure() {
+  auto sc = createSubscriptionContext();
+  sc->event_id = 32800;
+  subscribe(&OpenBSMSSHLoginSubscriber::Callback, sc);
+}
+
+Status OpenBSMSSHLoginSubscriber::Callback(
```
Comment (obelisk) [inline osquery/tables/events/freebsd/openbsm_events.cpp]: Does this work on FreeBSD?

File: osquery/tables/events/freebsd/openbsm_events.cpp [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+#include "osquery/events/darwin/openbsm.h"
+
+namespace osquery {
+
+namespace tables {
+extern long getUptime();
+}
+
+static inline void OpenBSM_AUT_SUBJECT32_EX(Row& r, const tokenstr_t& tok) {
```
Comment (obelisk) [inline osquery/tables/events/freebsd/openbsm_events.cpp]: Maybe we should pull these out into a header file, or some OpenBSM utils. We now have this function duplicated twice and parsing for more tokens would need to be copied into both files. Thoughts? I can handle the testing on macOS if you want to do that :)

File: osquery/tables/events/freebsd/openbsm_events.cpp (hunk 1 above)
Comment (lxcode) [reply osquery/tables/events/freebsd/openbsm_events.cpp]: Hm, perhaps fexecve (43144)? And maybe 43131, which is a MAC-related version of it. (MAC as opposed to DAC).

File: osquery/tables/events/freebsd/openbsm_events.cpp (hunk 2 above)
Comment (lxcode) [reply osquery/tables/events/freebsd/openbsm_events.cpp]: I don't see anything platform-dependent here — was there something that jumped out at you?

File: osquery/tables/events/freebsd/openbsm_events.cpp (hunk 3 above)
Comment (lxcode) [reply osquery/tables/events/freebsd/openbsm_events.cpp]: Yeah, I was kind of leaving the multi-platform deduplication decisions to one of the core team members, cause this isn't the only place we have that. TBH the symlink + couple ifdefs approach is pretty straightforward but I know people hate that. :)

File: osquery/tables/events/freebsd/openbsm_events.cpp (hunk 2 above)
Comment (obelisk) [reply osquery/tables/events/freebsd/openbsm_events.cpp]: This code that this listens too might not fire on FreeBSD

Comment (lxcode) [reply osquery/tables/events/freebsd/openbsm_events.cpp]: Yeah, should fire just fine, the event is there on all platforms.

**Comment by osqueryer** [comment]
:-1: The commit 91a4badffd140457c4a0168aeb586fa511197d45 (Job results: [2008](https://jenkins.osquery.io/job/osqueryPullRequestOSX/2008/)) failed one or more tests (**macOS/OS X**).

**Comment by directionless** [comment]
FreeBSD is on-hold. https://github.com/osquery/osquery/issues/6497 supercedes

**Pull request closed without merging**